adyen.php

```diff
@@ -606,21 +653,55 @@ public function hookPayment()
 
         $this->context->controller->addCSS($this->_path . 'css/adyen.css', 'all');
 
+        $amount = $this->context->cart->getOrderTotal();
+        $currency = $this->context->currency->iso_code;
+        $address = new Address($this->context->cart->id_address_invoice);
+        $isoAddress = Country::getIsoById($address->id_country);
+        $shopperReference = $this->context->cart->id_customer;
+        $shopperLocale = $this->helper_data->getLocale($this->context);
+
+        $payments = "";
+        $paymentMethods = $this->helper_data->fetchPaymentMethods($isoAddress, $amount, $currency, $shopperReference, $shopperLocale);
+        if (!$this->context->customer->is_guest && !empty($paymentMethods['oneClickPaymentMethods'])) {
+            $oneClickPaymentMethods = $paymentMethods['oneClickPaymentMethods'];
+            foreach ($oneClickPaymentMethods as $storedCard) {
+                if (!empty($storedCard["storedDetails"]["card"]["expiryMonth"])) {
+                    $this->context->smarty->assign(
+                        array(
+                            'locale' => $this->helper_data->getLocale($this->context),
+                            'originKey' => $this->helper_data->getOriginKeyForOrigin(),
+                            'environment' => Configuration::get('ADYEN_MODE'),
+                            'paymentProcessUrl' => $this->context->link->getModuleLink($this->name, 'Payment', array(),
+                                true),
+                            'threeDSProcessUrl' => $this->context->link->getModuleLink($this->name, 'ThreeDSProcess',
+                                array(), true),
+                            'prestashop16' => true,
+                            'oneClickPaymentMethod' => json_encode($storedCard),
+                            'recurringDetailReference' => $storedCard['recurringDetailReference'],
+                            'name' => $storedCard['name'],
+                            'number' => $storedCard['storedDetails']['card']['number']
+                        )
+                    );
+                }
+                $payments .= $this->display(__FILE__, '/views/templates/front/oneclick.tpl');
+            }
+        }
```

Comment:
This block doesn't look too different from the previous one. Can we extract a method to unify these?